distributed/scheduler.py

```diff
@@ -1371,9 +1407,12 @@ def __init__(
 
         self._transitions = {
             ("released", "waiting"): self.transition_released_waiting,
+            ("released", "speculative"): self.transition_waiting_speculative,
             ("waiting", "released"): self.transition_waiting_released,
             ("waiting", "processing"): self.transition_waiting_processing,
             ("waiting", "memory"): self.transition_waiting_memory,
+            ("waiting", "speculative"): self.transition_waiting_speculative,
+            ("speculative", "processing"): self.transition_speculative_processing,
```

Comment:
We'll eventually want to see what happens when a speculative task gets pushed back to released/waiting again, such as if a worker goes down.  The catch all behavior is, I think, to move things back to released.  I think that if you handle that that things should be ok, we may also want a speculative->waiting though.  Same with errors.  I suspect that when a processing task errs it will mark all of its dependents as errored, and so the scheduler will try to enact a speculative->erred transition.  

In some of these it may be that the current `self.transition_processing_*` methods will work without modification, but we'll probably have to verify this.